langchain4j-local-ai/src/test/java/dev/langchain4j/model/localai/AbstractLocalAiInfrastructure.java

```diff
@@ -0,0 +1,100 @@
+package dev.langchain4j.model.localai;
+
+import com.github.dockerjava.api.DockerClient;
+import com.github.dockerjava.api.command.InspectContainerResponse;
+import com.github.dockerjava.api.model.Image;
+import org.testcontainers.DockerClientFactory;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.utility.DockerImageName;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+public class AbstractLocalAiInfrastructure {
+
+    private static final String LOCAL_AI_IMAGE = "quay.io/go-skynet/local-ai:v1.40.0";
+
+    private static final String LOCAL_IMAGE_NAME = "tc-local-ai";
+
+    private static final String LOCAL_LOCAL_AI_IMAGE = String.format("%s:%s", LOCAL_IMAGE_NAME, DockerImageName.parse(LOCAL_AI_IMAGE).getVersionPart());
+
+    private static final List<String[]> CMDS = Arrays.asList(
+            new String[] {"curl", "-o", "/build/models/ggml-gpt4all-j", "https://gpt4all.io/models/ggml-gpt4all-j.bin"},
+            new String[] {"curl", "-o", "/build/models/ggml-model-q4_0.gguf", "https://huggingface.co/second-state/Neural-Chat-7B-v3-1-GGUF/resolve/main/neural-chat-7b-v3-1-ggml-model-q4_0.gguf"});
```

Comment:
`LocalAiEmbeddingModelIT#should_embed` fails with

```
{"error":{"code":500,"message":"endpoint disabled for this model by API configuration","type":""}}
```

Reply:
For `LocalAiEmbeddingModelIT` you need to download proper embedding model, e.g. `wget https://huggingface.co/skeskinen/ggml/resolve/main/all-MiniLM-L6-v2/ggml-model-q4_0.bin -O models/all-minilm-l6-v2` and then add `models/all-minilm-l6-v2.yaml` file with the following content:
```
name: all-minilm-l6-v2
parameters:
  model: bert
backend: bert-embeddings
embeddings: true
```
I am not sure this will work out of the box, I have troubles pulling `quay.io/go-skynet/local-ai:v1.40.0`

Reply:
I've updated the PR with those changes but unable to make it work. Do you use another image that I can try?

Reply:
Seems like some problem with that model, but I can't find the reason now...

Luckily, I had some old model on my PC that works, so I've uploaded it here: https://huggingface.co/LangChain4j/localai-embeddings/resolve/main/ggml-model-q4_0
You can also download yaml file directly from here: https://huggingface.co/LangChain4j/localai-embeddings/resolve/main/ggml-model-q4_0.yaml
You will need to rename it back to `ggml-model-q4_0`, sorry

Reply:
What error you are getting? Seems to work fine for me

Reply:
it is expected `hello` but the actual answer is `to me` or `to the world`.

Reply:
ah, this is not that important, we can also do `assertThat(answer).isNotBlank();`